knode/validator/validator.go

```diff
@@ -191,8 +193,14 @@ func (val *validator) SetCoinbase(walletAccount accounts.WalletAccount) error {
 	return nil
 }
 
-func (val *validator) SetDeposit(deposit *big.Int) {
+func (val *validator) SetDeposit(deposit *big.Int) error {
+	if val.Validating() {
```

Comment:
I think that this validation is not necessary anymore. we just call SetDeposit upon the validators' start method which verifies if the validator is running, or when we launch the client.